AsyncMediaServerProcessClientApp/asyncMediaServerProcessScript.php

```diff
@@ -1,73 +1,81 @@
 <?php
 
+//Define consts
+define("MAX_EXECUTION_ATTEMPTS", 5);
+
+define("FILE_STATUS_DONE", "done");
+define("FILE_STATUS_IN_PROGRESS", "in_progress");
+define("FILE_STATUS_ERROR", "error");
+define("FILE_STATUS_RETRY", "retry");
+
+//Read config file
 $configFilePath = $argv[1];
 $config = parse_ini_file($configFilePath);
 
-//Load configuration
-$files = glob($config['dirName'] . '/*.xml');
-if (!count($files))
+//Create REquired Directory For Script Execution
+if(!createDirIfNotExists($config['dirName'] . "/error/"))
 {
-	logMsg("No files at this time.");
 	die();
 }
 
-logMsg(count ($files) . " files found.");
-
-$inProgress = array();
-if (file_exists('in_progress'))
+if(!createDirIfNotExists($config['dirName'] . "/complete/"))
 {
-	$content = file_get_contents('in_progress');
-	$inProgress = explode('\n', $content);
+	die();
 }
 
-$complete = array();
-if (file_exists('complete'))
+//Load configuration
+$files = glob($config['dirName'] . '/*.xml');
+if (!count($files))
 {
-	$content = file_get_contents('complete');
-	$complete = explode('\n', $content);
+	logMsg("No new files found at this time.");
+	die();
 }
 
+logMsg(count ($files) . " files found.");
+
 //Sort files - ascending creation date
 foreach ($files as $f){
   $tmp[realpath($f)] = filectime($f);
 }
+
 asort($tmp);
 $files = array_keys($tmp);
-require_once 'client/KalturaClient.php';
 
+require_once 'client/KalturaClient.php';
 $clientConfig = new KalturaConfiguration();
 $clientConfig->serviceUrl = $config['serviceUrl'];
 $clientConfig->partnerId = $config['partnerId'];
 $client = new KalturaClient($clientConfig);
-$ks = $client->generateSessionV2($config['adminSecret'], '', KalturaSessionType::ADMIN, $clientConfig->partnerId, 86400, null);
+$ks = $client->generateSessionV2($config['adminSecret'], '', KalturaSessionType::ADMIN, $clientConfig->partnerId, 86400, "disableentitlement");
 $client->setKs($ks);
 
 foreach ($files as $f)
-{
-	if (in_array($f, $inProgress) || in_array($f, $complete))
-	{
-		continue;
-	}
-	
-	$inProgress[] = $f;
-	file_put_contents('in_progress', implode('\n', $inProgress));
-	
-	$handled = false;
-	while (!$handled)
+{	
+	$fileStatus = FILE_STATUS_IN_PROGRESS;
+	while ($fileStatus == FILE_STATUS_IN_PROGRESS)
```

Comment:
You don't need the while anymore, the results of handleUploadXMLResource is always FILE_STATUS_DONE, which also means that you don't have to check the returned value, or even to return a value.